auto-value-gson/src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java

```diff
@@ -289,13 +325,27 @@ public TypeSpec createTypeAdapter(ClassName className, ClassName autoValueClassN
       }
     }
 
+    ProcessingEnvironment processingEnvironment = context.processingEnvironment();
+    TypeMirror typeAdapterFactory = processingEnvironment
+            .getElementUtils()
+            .getTypeElement("com.google.gson.TypeAdapterFactory")
+            .asType();
+    Types typeUtils = processingEnvironment.getTypeUtils();
+
     ImmutableMap<Property, FieldSpec> nonparameterizedAdapters = filterParameterizedAdapters(adapters, false);
     for (Map.Entry<Property, FieldSpec> entry : nonparameterizedAdapters.entrySet()) {
       Property prop = entry.getKey();
       FieldSpec field = entry.getValue();
-      if (entry.getKey().type instanceof ParameterizedTypeName) {
+      if (prop.typeAdapter != null) {
+        if (typeUtils.isAssignable(prop.typeAdapter, typeAdapterFactory)) {
+          constructor.addStatement("this.$N = new $T().create($N, $L)", field, TypeName.get(prop.typeAdapter),
+                  gsonParam, makeType(prop.type));
```

Comment:
Should we use `TypeToken#get(Class)` instead of `new TypeToken<Class>(){}` for non parameterized types?